sys/psa_crypto/psa_persistent_storage/psa_crypto_persistent_storage.c

```diff
@@ -0,0 +1,127 @@
+    char string_path[strlen(vfs_mountpoints_xfa[0].mount_point) + UINT32_T_MAX_STRING_LEN];
+
+    sprintf(string_path, "%s/%d", vfs_mountpoints_xfa[0].mount_point, (int) id);
+
+    /* Check whether file already exists */
+    int fd = vfs_open(string_path, O_RDWR, 0);
+    if (fd >= 0) {
+        DEBUG("[psa_crypto] persist key: key with this ID already exists in storage: %d\n", fd);
+        vfs_close(fd);
+        return PSA_ERROR_ALREADY_EXISTS;
+    }
+
+    /* If file does not exist, create one */
+    fd = vfs_open(string_path, O_CREAT | O_RDWR, 0);
+
+    if (fd <= 0) {
+        DEBUG("[psa_crypto] persist key: Can not open file: %d\n", fd);
+        return PSA_ERROR_STORAGE_FAILURE;
+    }
+
+    if (vfs_write(fd, cbor_buf, cbor_buf_len) < 0) {
+        DEBUG("[psa_crypto] persist key: Can not write to file: %d\n", fd);
+        return PSA_ERROR_STORAGE_FAILURE;
+    }
+
+    if (vfs_close(fd) != 0) {
+        DEBUG("[psa_crypto] persist key: Can not close file: %d\n", fd);
+        return PSA_ERROR_STORAGE_FAILURE;
+    }
+
+    return PSA_SUCCESS;
+}
+
+psa_status_t psa_get_cbor_encoded_key_slot_from_file(psa_key_id_t id,
+                                                     uint8_t *cbor_buf,
+                                                     size_t cbor_buf_size,
+                                                     size_t *cbor_data_len)
+{
+    int uint32_t_max_string_len = 10;
+    char string_path[strlen(vfs_mountpoints_xfa[0].mount_point) + uint32_t_max_string_len];
+
+    sprintf(string_path, "%s/%d", vfs_mountpoints_xfa[0].mount_point, (int) id);
+
+    int fd = vfs_open(string_path, O_RDONLY, 0);
+
+    if (fd <= 0) {
+        DEBUG("[psa_crypto] read persisted key: Can not open file: %d\n", fd);
+        return (fd == -2 ? PSA_ERROR_DOES_NOT_EXIST : PSA_ERROR_STORAGE_FAILURE);
+    }
+
+    ssize_t cbor_bytes = vfs_read(fd, cbor_buf, cbor_buf_size);
+    if (cbor_bytes < 0) {
+        DEBUG("[psa_crypto] read persisted key: Can not read from file\n");
+        return (fd == -2 ? PSA_ERROR_DOES_NOT_EXIST : PSA_ERROR_STORAGE_FAILURE);
+    }
+
+    if (vfs_close(fd) != 0) {
+        DEBUG("[psa_crypto] read persisted key: Can not close file: %d\n", fd);
+        return PSA_ERROR_STORAGE_FAILURE;
+    }
+
+    *cbor_data_len = cbor_bytes;
+    return PSA_SUCCESS;
+}
+
+psa_status_t psa_destroy_persistent_key(psa_key_id_t key_id)
+{
+    if (psa_key_id_is_volatile(key_id)) {
+        DEBUG("[psa_crypto] persist key: ID is volatile\n");
+        return PSA_ERROR_INVALID_ARGUMENT;
+    }
+
+    /* read persisted key */
+    /* dir path needs to be as long as mount point + psa_key_id_t as string */
+    int uint32_t_max_string_len = 10;
+    char string_path[strlen(vfs_mountpoints_xfa[0].mount_point) + uint32_t_max_string_len];
+
+    sprintf(string_path, "%s/%d", vfs_mountpoints_xfa[0].mount_point, (int) key_id);
+
+    int fd = vfs_unlink(string_path);
```

Comment:
Isn't there any way to safer delete a file in vfs? maybe overwrite first with zeros?

Reply:
VFS does not provide a better way. Completely deleting the key requires to erase or overwrite the data.
Erasing is impractical, because usually you can only erase whole pages or sections.
Overwriting flash without erasing it, is not supported on all devices. 

The scope of this PR is mainly to provide a way to persistently store keys. Making keys unreadable or unrecoverable can be added in a follow-up step. 
I added a warning to the documentation instead.